src/Bicep.LangServer/Registry/ModuleRestoreScheduler.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Registry;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Bicep.LanguageServer.Registry
+{
+    public sealed class ModuleRestoreScheduler : IModuleRestoreScheduler, IDisposable
```

Comment:
Since we're using a `Task` for the background work, does it make sense here to implement `IAsyncDisposable` instead of `IDisposable`?

Reply:
I wasn't aware of that interface. Thanks for pointing it out! The main challenge was not being able to await inside the `Dispose()` implementation, so this should help.